crates/orchestrator/src/spawner.rs

```diff
@@ -158,11 +162,40 @@ where
     println!("🚀 {} : metrics link {prometheus_uri}", node.name);
     println!("📓 logs cmd: tail -f {}/{}.log", base_dir, node.name);
     println!("\n");
+
+    let client = retry(5, || async { OnlineClient::from_url(&ws_uri).await })
+        .await
+        .context(format!("Failed to connect to node rpc at {ws_uri}"))?;
+
+    let rpc = RpcClient::from_url(&ws_uri)
+        .await
+        .context(format!("Failed to connect to rpc client at {ws_uri}"))?;
+
     Ok(NetworkNode::new(
         node.name.clone(),
         ws_uri,
+        rpc,
+        client,
         prometheus_uri,
         node.clone(),
         running_node,
     ))
 }
+
+async fn retry<T, F, R, E>(retry_count: u32, connect: F) -> Result<R, E>
+where
+    T: Future<Output = Result<R, E>>,
+    F: Fn() -> T,
+{
+    let mut failed_count = 0;
+    loop {
+        match connect().await {
+            Err(_) if retry_count > failed_count => {
+                failed_count += 1;
+                println!("Error connecting, retrying {failed_count}/{retry_count} ...");
```

Comment:
Logging is on our `//TODO` list, we can add here a comment to easily update with when we start moving all those println to the logger impl.